gamemodes/attack_of_the_mimics/entities/weapons/swep_aotm_claws.lua

```diff
@@ -174,31 +179,61 @@ function SWEP:SecondaryAttack()
 	self:SetNextPrimaryFire( CurTime() + self.Secondary.Delay )
 	self:SetNextSecondaryFire( CurTime() + self.Secondary.Delay )
 	
+	if not IsValid(self.Owner) then return end
+	
+	local owner = self.Owner
+	
 	if SERVER then
-		self.Owner:EmitSound("npc/fast_zombie/fz_scream1.wav", 95, Lerp(math.random(), 80, 100))
+		owner:LagCompensation(true)
 		
-		self.Owner:SetEnergy(99)
+		local spos = self.Owner:GetShootPos()
+		local sdest = spos + (self.Owner:GetAimVector() * 70)
+
+		local kmins = Vector(1,1,1) * -10
+		local kmaxs = Vector(1,1,1) * 10
+
+		local tr = util.TraceHull({start=spos, endpos=sdest, filter=self.Owner, mask=MASK_SHOT_HULL, mins=kmins, maxs=kmaxs})
+
+		-- Hull might hit environment stuff that line does not hit
+		if not IsValid(tr.Entity) then
+			tr = util.TraceLine({start=spos, endpos=sdest, filter=self.Owner, mask=MASK_SHOT_HULL})
+		end
+
+		local hitEnt = tr.Entity
 		
-		self:SetScreamReady(false)
-		self.next_scream_ready = CurTime() + 60
+		if IsValid(hitEnt) and not hitEnt:IsWorld() then
+			local mimic_body = owner:GetMimicBody()
+			
+			if hitEnt:IsPlayer() and hitEnt:Team() == TEAM_MIMIC then
+				mimic_body:Mimic(hitEnt:GetMimicBody():GetModel(), hitEnt:GetMimicBody():GetSkin())
+			else
+				mimic_body:Mimic(hitEnt:GetModel(), hitEnt:GetSkin())
+			end
+		end
+		
+		owner:LagCompensation(false)
 	end
 end
 
 
 function SWEP:CanSecondaryAttack()
 	if not IsValid(self.Owner) then return end
 	
-	return self:GetScreamReady()
+	return true
 end
 
 
-function SWEP:DrawHUD()
-	if not self:GetScreamReady() then
-		surface.SetDrawColor(Color(255,0,0))
+function SWEP:Reload()
+	if not self:GetScreamReady() then return end
+	
+	if SERVER then
+		self.Owner:EmitSound( table.Random(self.Scream.Sounds), 95, Lerp(math.random(), 80, 100) )
```

Comment:
Again, this was mentioned in the wiki to be less efficient. https://wiki.garrysmod.com/page/table/Random